enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.v1.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  SdkIframeEmbedSettings,
+  SdkIframeEmbedTagMessage,
+  SdkIframeEmbedTagSettings,
+} from "./types/embed";
+
+const EMBEDDING_ROUTE = "embed/sdk/v1";
+
+type EmbedSettingKey = keyof SdkIframeEmbedSettings;
+
+const ALLOWED_EMBED_SETTING_KEYS = [
+  "apiKey",
+  "instanceUrl",
+  "dashboardId",
+  "questionId",
+  "template",
+  "theme",
+  "locale",
+] as const satisfies EmbedSettingKey[];
+
+type AllowedEmbedSettingKey = (typeof ALLOWED_EMBED_SETTING_KEYS)[number];
+
+class MetabaseEmbed {
+  static readonly VERSION = "1.0.0";
+
+  private _settings: SdkIframeEmbedTagSettings;
+  private _isEmbedReady: boolean = false;
+  private iframe: HTMLIFrameElement | null = null;
+
+  constructor(settings: SdkIframeEmbedTagSettings) {
+    if (!settings.apiKey || !settings.instanceUrl) {
```

Comment:
How does this check relate to https://github.com/metabase/metabase/blob/d908c8af7f07698cd5b367f6246d52b0bb6a8bac/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L21

At glance, it feels like this check would never be true since we guard check before rendering this component.

Reply:
- The `if (!settings.apiKey || !settings.instanceUrl) {` in the embed.js is for validation purposes.
- The `if (!embedSettings || !embedSettings.instanceUrl || !embedSettings.apiKey) {` in the iframe for data loading purposes -- `embedSettings` will be `null` for a long while until the `postMessage` communication completes (the iframe tells the script tag "we're ready", and the script tag sending the embed settings to the iframe).

Reply:
Clarified as comments in https://github.com/metabase/metabase/pull/56915/commits/485a26e6adefdad649db69e94ecca1289965216f